src/lib.rs

```diff
@@ -20,6 +21,23 @@ use std::convert::TryInto;
 
 include!(concat!(env!("OUT_DIR"), "/lib.rs"));
 
+/// Wrap IronOxide to allow namespacing of the advanced operations
+struct IronSdkAdvanced<'a>(&'a IronOxide);
+
+impl<'a> IronSdkAdvanced<'a> {
+    pub fn advanced(ironoxide: &IronOxide) -> IronSdkAdvanced {
+        IronSdkAdvanced(ironoxide)
+    }
+
+    pub fn document_encrypt_unmanaged(
+        &self,
+        data: &[i8],
+        opts: &DocumentEncryptOpts,
+    ) -> Result<DocumentEncryptUnmanagedResult, String> {
+        Ok(self.0.document_encrypt_unmanaged(i8_conv(data), opts)?)
```

Comment:
You unnecessarily unwrapping a Result and re-wrapping immediately.

Reply:
The re-wrap allows mapping of the error type for us